sklearn/linear_model/randomized_l1.py

```diff
@@ -354,6 +354,11 @@ def _randomized_logistic(X, y, weights, mask, C=1., verbose=False,
         X *= (1 - weights)
 
     C = np.atleast_1d(np.asarray(C, dtype=np.float64))
+    if C.ndim > 1:
+        raise ValueError("C should be 1-dimensional array, "
+                         "but got a %d-dimensional array instead: %d."
```

Comment:
This is going to give an error because `"%d"` is not a valid format for an array ... this is why we need a test.

Please use `.format` rather than `%` (less chance of making silly mistakes like this). Also please use "array-like" and not "array" as mentioned in one of my previous comment.